src/test/scala/chiselTests/properties/ObjectSpec.scala

```diff
@@ -94,4 +94,95 @@ class ObjectSpec extends ChiselFlatSpec with MatchesAndOmits {
       "propassign obj1.in, in"
     )()
   }
+
+  it should "support Instances of Objects in Class ports" in {
+    val chirrtl = ChiselStage.emitCHIRRTL(
+      new RawModule {
+        Definition({
+          val cls1 = Definition(new Class {
+            override def desiredName = "Test"
+            val in = IO(Input(Property[Int]()))
+            val out = IO(Output(Property[Int]()))
+            out := in
+          })
+
+          val cls2 = Definition(new Class {
+            override def desiredName = "Test"
+            val in = IO(Input(Property[Int]()))
+            val out = IO(Output(Property[Int]()))
+            out := in
+          })
+
+          new Class {
+            override def desiredName = "Parent"
+            val out1 = IO(Output(Class.getReferenceType(cls1)))
+            val out2 = IO(Output(Class.getReferenceType(cls2)))
+
+            val obj1 = Instance(cls1)
+            val obj2 = Instance(cls2)
+
+            val inst1 = Class.getObjectInstance(obj1)
+            val inst2 = Class.getObjectInstance(obj2)
+
+            out1 := inst1.getReference
+            out2 := inst2.getReference
+          }
+        })
+      }
+    )
+
+    matchesAndOmits(chirrtl)(
+      "class Parent",
+      "output out1 : Inst<Test>",
+      "output out2 : Inst<Test_1>",
+      "object obj1 of Test",
+      "object obj2 of Test_1",
+      "propassign out1, obj1",
+      "propassign out2, obj2"
+    )()
+  }
+
+  it should "support Instances of Objects in Module ports" in {
+    val chirrtl = ChiselStage.emitCHIRRTL(
+      new RawModule {
+        val cls1 = Definition(new Class {
+          override def desiredName = "Test"
+          val in = IO(Input(Property[Int]()))
+          val out = IO(Output(Property[Int]()))
+          out := in
+        })
+        val cls2 = Definition(new Class {
+          override def desiredName = "Test"
+          val in = IO(Input(Property[Int]()))
+          val out = IO(Output(Property[Int]()))
+          out := in
+        })
+
+        Module(new RawModule {
+          override def desiredName = "Parent"
+          val out1 = IO(Output(Class.getReferenceType(cls1)))
+          val out2 = IO(Output(Class.getReferenceType(cls2)))
+
+          val obj1 = Instance(cls1)
+          val obj2 = Instance(cls2)
+
+          val inst1 = Class.getObjectInstance(obj1)
+          val inst2 = Class.getObjectInstance(obj2)
+
+          out1 := inst1.getReference
+          out2 := inst2.getReference
+        })
+      }
+    )
+
+    matchesAndOmits(chirrtl)(
+      "module Parent",
+      "output out1 : Inst<Test>",
+      "output out2 : Inst<Test_1>",
+      "object obj1 of Test",
+      "object obj2 of Test_1",
+      "propassign out1, obj1",
+      "propassign out2, obj2"
+    )()
```

Comment:
What are `inst1` and `inst2`? We don't really have a similar concept for normal D/I. Should we instead just make connections support `Instance[A <: Class]`? Or perhaps we should instead make the method to get the object instance be a method on `Instance[A <: Class]` rather than on `object Class`?

Reply:
Hmm yes this is a good point. This certainly feels a little weird, and it would be great if we could directly connect the `Instance[A <: Class]`.

I implemented this a while ago and just sent the PR... trying to page in what led me to this. Looking at what I am doing in getObjectInstance, I guess we need a way to go from a `Instance[A <: Class]` to a `Property[ClassType]`, and add the appropriate binding for the property. So basically, I think whatever getObjectInstance+getReference are doing, we would do internally within `connect`. I can try some experiments.

Reply:
I also like your other idea as a backup, we could define a `getReference` method on `Instance[A <: Class]` to do this bookkeeping. I'll try to see if I can make it work to directly connect `Instance[A <: Class]` first.

Reply:
Note that for either case you probably need to be using extension methods (aka `implicit class`), eg. see `Connectable`.

Reply:
Makes sense, I'll poke around with it.

Reply:
Yep, the first idea works. I can add an extension method to `Property[ClassType]` that adds a `:=` method that accepts an Instance of some class. I need to spend a little more time playing with this to see if we can completely get rid of this StaticObject thing or if it's needed internally for bookkeeping. I'm also hoping we could statically enforce that the Instance is of the right class... at the least we can do a runtime check.

Reply:
I don't see a way to get rid of StaticObject, since we need something to hold a reference to the instance's underlying module clone until after its been named, so we can set the ref for the Property[ClassType] we make from the instance to the module clone's ref. I also don't know the best way to enforce the types match statically, since we need the module clone's `name`, which I don't know how to bake that into a Scala type.

Reply:
That's interesting! I originally thought it only can be done in Scala 3.